Unix/http/httpauth.c

```diff
@@ -1605,83 +1612,69 @@ void Deauthorize(_In_ Http_SR_SocketData * handler)
-    Strlcat(file, "/sockets/", PAL_MAX_PATH_SIZE);
-    Strlcat(file, rString, PAL_MAX_PATH_SIZE);
+    Http_SR_SocketData *handler = (Http_SR_SocketData*)(uintptr_t)msg->handle;
+    HttpHeaders *headers = &handler->recvHeaders;
 
-    return 0;
-}
+    if (!msg->result)
+    {
+        handler->httpErrorCode = HTTP_ERROR_CODE_UNAUTHORIZED;
+        auth_response = (char *)RESPONSE_HEADER_UNAUTH_FMT;
+        response_len  = RESPONSE_HEADER_UNAUTH_FMT_LEN;
 
-static int _AskServerToAuthenticateUser(const char *user,
-                                        const char *password,
-                                        Http_SR_SocketData *handler)
-{
-    int fd;
-    int r;
-    char file[PAL_MAX_PATH_SIZE];
+        trace_HTTP_UserAuthFailed("server determined that user is not authorized");
+        handler->authFailed = TRUE;
 
-    trace_AskServerToAuthenticate();
+        _SendAuthResponse(handler, auth_response, response_len);
+        goto Done;
+    }
 
-    /* Verify if user is in cache already */
-    if (0 == CredCache_CheckUser(user, password))
-        return 0;
+    /* Add user to cache if auth was ok */
+    CredCache_PutUser(headers->username, headers->password);
 
-    r = _GeneratePipeFile(file);
-    if (0 != mkfifo(file, 0400))
+    if (0 != LookupUser(headers->username, &handler->authInfo.uid, &handler->authInfo.gid))
     {
-        return -1;
-    }
+        trace_GetUserUidGid_Failed(headers->username);
+        
+        handler->httpErrorCode = HTTP_ERROR_CODE_UNAUTHORIZED;
+        auth_response = (char *)RESPONSE_HEADER_UNAUTH_FMT;
+        response_len  = RESPONSE_HEADER_UNAUTH_FMT_LEN;
 
-    r = AskServerToAuthenticate(file, user, password, &handler->engineBatch, 
-                                handler->http->selector);
-    if (0 != r)
-    {
-        unlink(file);
-        return -1;
+        trace_HTTP_UserAuthFailed("basic auth user creds not present");
+        handler->authFailed = TRUE;
+        _SendAuthResponse(handler, auth_response, response_len);
+        goto Done;
     }
 
-    if ((fd = open(file, O_RDONLY)) < 0)
-    {
-        unlink(file);
-        return -1;
-    }
+    handler->httpErrorCode = 0; // Let the request do the error code
+    handler->isAuthorised = TRUE;
 
-    if (sizeof(int) != ReadFile(fd, &r, sizeof(int)))
+    r = Process_Authorized_Message(handler);
+    if (MI_RESULT_OK != r)
     {
-        trace_UserAuth_FailedToRead( errno );
-        r = -1;
-        goto done;
+        return MI_RESULT_FAILED;
     }
 
-done:
-    close(fd);
-    unlink(file);
+Done:
+    handler->recvPage = 0;
+    handler->receivedSize = 0;
+    memset(&handler->recvHeaders, 0, sizeof(handler->recvHeaders));
+    handler->recvingState = RECV_STATE_HEADER;
 
-    /* Add user to cache if auth was ok */
-    if (0 == r)
-        CredCache_PutUser(user, password);
-
-    return r;
+    // Resume selector processing of this socket
+    handler->handler.mask |= SELECTOR_READ;
```

Comment:
move right before Process_Authorized_Message. Not set for the error cases as they set the write state to send the error. Also, Process_Authorized_Message will ultimately set the write state to send the response back